lib/postgrex/replication.ex

```diff
@@ -325,23 +388,9 @@ defmodule Postgrex.Replication do
 
   @doc false
   def connect(_, s) do
-    opts = opts()
-
-    case Protocol.connect([types: nil] ++ opts()) do
+    case Protocol.connect(opts()) do
       {:ok, protocol} ->
-        s = %{s | protocol: protocol}
-        {create, start} = commands(opts)
-
-        with {:ok, %Postgrex.Result{}, protocol} <- Protocol.handle_simple(create, protocol),
-             {:ok, protocol} <- Protocol.handle_replication(start, protocol),
-             {:ok, protocol} <- Protocol.checkin(protocol) do
-          {:ok, %{s | protocol: protocol}}
-        else
-          # If we can't start replication, we assume that auto_reconnect can't
-          # solve it either, so we just raise the error as a readable message.
-          {_error, reason, _protocol} ->
-            raise reason
-        end
+        {:ok, %{s | protocol: protocol}}
```

Comment:
I think `auto_reconnect` will no longer work as expected now. I think that, if replication has started, we should store the replication options in the connection state and automatically restart replication. WDYT?

Reply:
Thank you, good point!. I added this in with some extra details:

- if the slot is temporary then it also must be recreated on restart
- added a warning about duplicate WAL messages on restart
- If replication can't be restarted then I raise an error, same as what was there before. Since I believe in this case the error is unexpected and requires investigation.

Reply:
Actually this is trickier than I originally thought because you can create a slot that you're not starting replication on. 

What do you think about not trying to recreate the slot and putting a note in the doc stating that replication on temporary slots can't be automatically restarted?

Reply:
> What do you think about not trying to recreate the slot and putting a note in the doc stating that replication on temporary slots can't be automatically restarted?

Sounds like a good way to go for now!